Add when-to-use-general section (#27)
cbor-wg/draft-ietf-cbor-serialization: draft-ietf-cbor-serialization.md

Ordinary serialization ({{OrdinarySerialization}}) satisfies the vast majority of CBOR use cases; therefore, the need for general serialization is rare and arises only in unusual circumstances.
The following are representative examples:

* Enable on-the-fly, streaming encoding of strings, arrays, and maps with indefinite lengths.
This is useful when an array, map, or string is many times larger than the available memory on the encoding device.

* Directly encode or decode integer values from hardware registers with fixed-size integer encoding.
CBOR is sufficiently simple that encoders and decoders for some protocols can be implemented solely in hardware without any software.
Fixed-size integer encoding allows values to be copied directly in and out of hardware registers.

* Enable in place update of the lengths of strings, arrays and maps by using fixed-size encoding of their lengths.
For example, if the length of a string is always encoded in 32 bits, increasing the length from 2^16 to 2^16+1, requires only overwriting the length field rather than shifting all 2^16 bytes of content.

* Transmission of non-trivial NaNs in floating-point values (see {{NaN}}).

> With only the exception of non-trivial NaNs, general serialization can encode all the same data types and value ranges as other serializations.

Reviewer: ```suggestion
Except for non-trivial NaNs, the other serializations can encode the same data types and value ranges as general serialization.
```
